### PR TITLE
Websocket codes

### PR DESCRIPTION
**Pull request by LordOfPolls**
## Pull Request Type

- [x] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description

Creating on behalf of @zevaryx. 
This PR marks additional gateway close codes as recoverable/resumable. 

### to be clear ###
If your bot is online 24/7 you **need** a process manager like docker, systemd or whatever. Do NOT rely on the library to automatically restart itself. The library certainly tries to recover, but the internet is a scary place 

## Changes

- As above
- Adds a sanity check for development to ensure resumable is a subset of recoverable 

## Python Compatibility

- [ ] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`

## Checklist

- [x] I've run the `pre-commit` code linter over all edited files
- [x] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [ ] I've updated / added  documentation, where applicable

### CODE REVIEW

File: interactions/client/const.py [hunk 1]
```diff
@@ -220,5 +220,49 @@ class MentionPrefix(Sentinel):
...
+NON_RESUMABLE_WEBSOCKET_CLOSE_CODES = (  # Codes that are recoverable, but the bot won't be able to resume the session
+    1000,  # Normal closure
+    1003,  # Unsupported Data
+    1008,  # Policy Violation
+    1009,  # Message too big
+    1011,  # Server error
+    1012,  # Server is restarting
+    1014,  # Handshake failed
+    1015,  # TLS error
+    4007,   # Invalid seq
```
Comment (AlbertUnruh) [inline interactions/client/const.py]:
```suggestion
    4007,  # Invalid seq
```

File: interactions/client/const.py [hunk 2]
```diff
@@ -220,5 +220,49 @@ class MentionPrefix(Sentinel):
 LIB_PATH = os.sep.join(__file__.split(os.sep)[:-2])
 """The path to the library folder."""
 
-RECOVERABLE_WEBSOCKET_CLOSE_CODES = (4000, 4001, 4002, 4003, 4005, 4007, 4008, 4009)
-NON_RESUMABLE_WEBSOCKET_CLOSE_CODES = (1000, 4007)
+# fmt: off
+RECOVERABLE_WEBSOCKET_CLOSE_CODES = (  # Codes that can are recoverable, and the bot will reconnect
```
Comment (AlbertUnruh) [inline interactions/client/const.py]: "can are", shouldn't it just be "are"?

Comment (AlbertUnruh) [reply interactions/client/const.py]:
```suggestion
RECOVERABLE_WEBSOCKET_CLOSE_CODES = (  # Codes that are recoverable, and the bot will reconnect
```

**Comment by AlbertUnruh** [comment]
besides my suggestions it looks good to me, *but I won't approve it since I don't know the codes and whether they are correct or not*

**Comment by LordOfPolls** [comment]
Merging without approval due to importance of this PR

**Pull request merged**